easier_net/easier_net.py

```diff
@@ -0,0 +1,174 @@
+        input_filter_layer=True, #Scales the inputs by parameter β
+        n_layers=2, #Number of hidden layers
+        n_hidden=10, #Number of hidden nodes in each layer
+        n_estimators=5, #Size of ensemble; number of SIER-nets being ensembled.
+        max_iters=40, #Number of epochs to run Adam     
+        max_prox_iters=0, #Number of epochs to run batch proximal gradient descent
+        input_pen=0, #λ1; Controls the input sparsity         
+        full_tree_pen=0.001, #λ2; controls the number of active layers and hidden nodes 
+        num_batches=3, #number of mini-batches for Adam
+        model_fit_params_file=None, #A json file that specifies what the hyperparameters are. If given, this will override the arguments passed in.
+    ):
+
+        self.num_classes=num_classes
+        self.input_filter_layer=input_filter_layer
+        self.n_layers=n_layers
+        self.n_hidden=n_hidden
+        self.n_estimators=n_estimators
+        self.max_iters=max_iters
+        self.max_prox_iters=max_prox_iters
+        self.full_tree_pen=full_tree_pen
+        self.input_pen=input_pen
+        self.num_batches=num_batches
+        self.model_fit_params_file=model_fit_params_file
+        self.score_criterion = (
+            nn.CrossEntropyLoss() if self.num_classes >= 2 else nn.MSELoss()
+        )       
+        if self.model_fit_params_file is not None:
+            self.estimators: List[sier_net.SierNetEstimator] = self.load_model(model_fit_params_file) 
+        else:
+            self.estimators: List[sier_net.SierNetEstimator] = self._generate_estimators()
+
+        assert num_classes != 1
+        assert input_filter_layer
+    
+    def _generate_estimators(self):
+        estimators=[]
+        for est in range(self.n_estimators): #TODO: i -> est?
+            est=sier_net.SierNetEstimator(
+                num_classes=self.num_classes,
+                input_filter_layer=self.input_filter_layer,
+                n_hidden= self.n_hidden,
+                max_iters=self.max_iters,
+                max_prox_iters=self.max_prox_iters,
+                full_tree_pen=self.full_tree_pen,
+                input_pen=self.input_pen,
+            ) 
+            
+            estimators.append(est)
+
+        return estimators
+
+
+    def write_model(self, filename): #save and write model to file
+        #save as .pt file
+        meta_state_dict = self.get_params()
+        meta_state_dict["state_dicts"] = [
+            est.net.state_dict() for est in self.estimators
+        ]
+        torch.save(meta_state_dict, filename)
+
+    def load_model(self, model_fit_params_file):
+        estimators = []
+        state_dict_list = torch.load(model_fit_params_file, map_location='cpu') #to avoid gpu ram surge
+        for state_dict in state_dict_list:
+            sier_net = sier_net.SierNetEstimator(
+                n_inputs=self.n_inputs,
```

Comment:
I feel like there are other things missing from here, like you never pass the argument for `n_layers`